delivery_carrier_label_batch/models/stock_batch_picking.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from openerp import _, api, fields, models
+
+
+class StockBatchPicking(models.Model):
+
+    """ Add carrier and carrier options on batch
+
+    to be able to massively set those options on related picking.
+
+    """
+    _inherit = "stock.batch.picking"
+
+    carrier_id = fields.Many2one(
+        'delivery.carrier', 'Carrier',
+        states={'done': [('readonly', True)]})
+    option_ids = fields.Many2many(
+        'delivery.carrier.option',
+        string='Options')
+
+    @api.multi
+    def action_set_options(self):
+        """ Apply options to picking of the batch
+
+        This will replace all carrier options in picking
+
+        """
+        for rec in self:
+            options_datas = {
+                'carrier_id': rec.carrier_id.id,
+                'option_ids': [(6, 0, rec.option_ids)],
+            }
+            rec.related_picking_ids.write(options_datas)
```

Comment:
The field `related_picking_ids` does not exist on the `stock.batch.picking` model ; it should be either `picking_ids` or `active_picking_ids`.